src/app.css

```diff
@@ -25,10 +27,56 @@
   h3,
   h4,
   h5,
-  h6,
-  titles,
-  labels {
-    font-family: theme('fontFamily.secondary');
+  h6 {
+    font-weight: theme('fontWeight.medium');
+  }
+
+  h1 {
+    font-size: theme('fontSize.2xl');
+
+    @media (min-width: theme('screens.md')) {
+      font-size: theme('fontSize.3xl');
+    }
+  }
+
+  h2 {
+    font-size: theme('fontSize.xl');
+
+    @media (min-width: theme('screens.md')) {
+      font-size: theme('fontSize.2xl');
+    }
+  }
+
+  h3 {
+    font-size: theme('fontSize.lg');
+
+    @media (min-width: theme('screens.md')) {
+      font-size: theme('fontSize.xl');
+    }
+  }
+
+  h4 {
+    font-size: theme('fontSize.base');
+
+    @media (min-width: theme('screens.md')) {
+      font-size: theme('fontSize.lg');
+    }
+  }
+
+  h5 {
+    font-size: theme('fontSize.sm');
+
+    @media (min-width: theme('screens.md')) {
+      font-size: theme('fontSize.base');
+    }
+  }
+
+  h6 {
+    font-size: theme('fontSize.xs');
+
+    @media (min-width: theme('screens.md')) {
+      font-size: theme('fontSize.sm');
+    }
   }
 
```

Comment:
Might be nice to add some helper classes here for emphasized or other types of frequently used text styles, but not a blocker for this PR.